subtitles.md

```diff
@@ -677,3 +677,36 @@ in segment 7.2 "Syntax and semantics of the subtitling segment" of ETSI EN 300 7
 Each Matroska Block **SHOULD** have a Duration indicating how long the DVB Subtitle Segments
 in that Block **SHOULD** be displayed.
 
+## ARIB (ISDB) subtitles
+
+The specifications for the ARIB B-24 subtitle bitstream format (short: ARIB subtitles)
+and its storage in MPEG transport streams can be found in the documents
+"ARIB STD-B24 - Data Coding and Transmission Specification for Digital Broadcasting",
+"ARIB STD-B10 - Service Information for Digital Broadcasting System",
+and "ARIB TR-B14 - Operational Guidelines for Digital Terrestrial Television Broadcasting".
+
+### Storage of ARIB subtitles
+
+#### CodecID
+
+The CodecID to use is `S_ARIBSUB`.
+
+#### CodecPrivate
+
+The CodecPrivate element is three bytes long and has the following structure:
+
+*    1 byte: component tag (bit string, left bit first)
+*    2 bytes: data component ID (bit string, left bit first)
+
+The semantics of the component tag are the same as those described in ARIB STD-B10, part 2, Annex J.
+The semantics of the data component ID are the same as those described in ARIB TR-B14, fascicle 2, Vol. 3, Section 2, 4.2.8.1.
+
+#### Storage of ARIB subtitles in Matroska Blocks
+
+Each Matroska Block consists of one or more ISDB Caption Data Groups as described
+in chapter 9 "Transmission of caption and superimpose" of ARIB STD-B24 volume 1 part 3.
```

Comment:
It sounds odd to have more than one data group in one Block, and by that I think you mean a frame. Does a "caption data group" just means it's a series of character ? Or they can be different languages ?

Reply:
"Caption data group" is defined in ARIB STD-B24 volume 1 part 3 chapter 9; an MPEGTS PES packet contains a "caption data", which consists of one or more "caption data groups", which are defined structures. I don't know why they call them that, but it's their terminology, not mine.

Technically different groups can be in different languages, but in practice there seems to always be one, with separate languages being separate PES streams (this is recommended by the B24 spec). The groups can be multiple caption statements, or a mix of "management" data (which configures display modes and such) and caption statements.

Reply:
Also, I got the "Each Matroska Block consists of one or more [...]" language from the DVB section of this file.

Reply:
A Matroska track cannot have multiple language. If there is more than one language then it needs to be split, or set to `undefined` language, which sucks when it comes to selecting proper subtitles.

There should be a description on how to handle the language mapping (they probably have some code, that have to be mapped to our strings) and what to do in case there's more than one language involved.

I know others don't do it, that's the legacy of what we had so far. But going forward we should do things correctly, at least for new things. For audio and video codecs there are a lot of fields that need to be mapped to Matroska (see the AV1 spec) at least for subtitles that's just the language and what the extra data contain.

Reply:
I could put a line in here that says that all data groups that contain caption statements must have the same language ID (or specifically language ID 0)? In practice, I doubt anything will enforce that (particularly since multilingual streams seem to just be a spec quirk that doesn't actually get used in the field), but I can stick it in there to make it clear that players don't have to support it (ffmpeg doesn't even support it in mpegts).

Reply:
That would be nice. At least if the case happens that means the file is bogus, and coders don't have to go to great length to make it work. Or just by reading the specs wonder how they are supposed to handle such a case.

Reply:
Asked around regarding if anyone had seen the usage of two subtitle languages in a single PID.

> I have added support for switching between the two available languages in my playback plugin, but I have not seen this being utilized.

> There are rumors regarding the feature being utilized by Disney Channel, but I've not been able to verify this.

So this is kind of similar to US style closed captions or DVB teletext, where you can have multiple different "alternatives" within a single stream, but at least its use seems to be quite rare.

Reply:
The last edit where it says it should not happen seems OK. If it happens, people are on their own to deal with it, ie split the tracks.